source/create.js

```diff
@@ -31,6 +35,24 @@ const create = defaults => {
 		handler: defaults.handler
 	});
 
+	got.merge = (...args) => {
+		const lastArgument = args[args.length - 1];
+		let methods;
+		let instances;
+
+		if (is.array(lastArgument)) {
+			methods = lastArgument;
+			instances = args.slice(0, -1);
+		} else {
+			methods = args[0].defaults.methods; // eslint-disable-line prefer-destructuring
+			instances = args;
+		}
+
+		instances = instances.length === 1 ? [got, ...instances] : instances;
```

Comment:
This hack enables merging two instances by `instanceA.merge(instanceB)`. You can always do `got.merge(instanceA, instanceB)`.

Reply:
That what I commented I don't think we should do in https://github.com/sindresorhus/got/pull/510#discussion_r208347443 Better to have one way to do something. Pick one of those, not both.

Reply:
Ok.